src/main/index.ts

```diff
@@ -97,11 +98,43 @@ async function start(bootBg: boolean) {
   }
 }
 
+
+if (process.defaultApp) {
+  if (process.argv.length >= 2) {
+    app.setAsDefaultProtocolClient('urbit-port', process.execPath, [path.resolve(process.argv[1])])
+  }
+} else {
+  app.setAsDefaultProtocolClient('urbit-port')
+}
+
+const gotTheLock = app.requestSingleInstanceLock()
+
+if (!gotTheLock) {
+  app.quit()
+  return;
+}
+
+app.on('second-instance', (event, commandLine, workingDirectory) => {
+  // Someone tried to run a second instance, we should focus our window.
+  if (mainWindow) {
+    if (mainWindow.isMinimized()) {
+      mainWindow.restore()
+    }
+    mainWindow.focus()
+  }
+})
+
 // This method will be called when Electron has finished
 // initialization and is ready to create browser windows.
 // Some APIs can only be used after this event occurs.
 app.on('ready', () => start(true));
 
+app.on('open-url', (event, url) => {
+  let ticket = url.split('urbit-port://')[1];
```

Comment:
This should follow the format urbit-port://incoming-id/{UUID}?key={KEY}&patp={PATP} based off the protocol. It's ok if UUID is blank which likely it would be in this case. I'm also open to another property like invite or ticket that just has the key + patp like bridge uses for the invite link.